src/ServiceControl/Monitoring/MonitoringDataPersister.cs

```diff
@@ -0,0 +1,101 @@
+﻿namespace ServiceControl.Monitoring
+{
+    using System.Linq;
+    using Raven.Client;
+    using ServiceControl.Contracts.HeartbeatMonitoring;
+    using ServiceControl.EndpointControl;
+    using ServiceControl.EndpointControl.Contracts;
+    using ServiceControl.Infrastructure;
+    using ServiceControl.Infrastructure.DomainEvents;
+
+    public class MonitoringDataPersister : IDomainHandler<HeartbeatingEndpointDetected>,
```

Comment:
These persistence operations should probably all be done with a patch. Only Monitored and HostName are really variable. Endpoint Name and Host Id are invariant as they're used to calculate the Id.